src/Facebook/HttpClients/HttpClientsFactory.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Copyright 2014 Facebook, Inc.
+ *
+ * You are hereby granted a non-exclusive, worldwide, royalty-free license to
+ * use, copy, modify, and distribute this software in source code or binary
+ * form for use in connection with the web services and APIs provided by
+ * Facebook.
+ *
+ * As with any software that integrates with the Facebook platform, your use
+ * of this software is subject to the Facebook Developer Principles and
+ * Policies [http://developers.facebook.com/policy/]. This copyright notice
+ * shall be included in all copies or substantial portions of the software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+namespace Facebook\HttpClients;
+
+use GuzzleHttp\Client;
+use InvalidArgumentException;
+
+class HttpClientsFactory
+{
+    private function __construct()
+    {
+        // a factory constructor should never be invoked
+    }
+
+    /**
+     * HTTP client generation.
+     * 
+     * @param FacebookHttpClientInterface|Client|string|null $handler
+     *
+     * @throws Exception                If the cURL extension or the Guzzle client aren't available (if required).
+     * @throws InvalidArgumentException If the http client handler isn't "curl", "stream", "guzzle", or an instance of Facebook\HttpClients\FacebookHttpClientInterface.
+     * 
+     * @return FacebookHttpClientInterface
+     */
+    public static function create($handler)
+    {
+        if (!$handler) {
+            return self::detectDefaultClient();
+        }
+
+        if (is_object($handler) && $handler instanceof FacebookHttpClientInterface) {
+            return $handler;
+        }
+
+        if (is_string($handler) && 'stream' === $handler) {
```

Comment:
this condition can be simplified as ``'stream' === $handler``, with the same behavior, but a smaller cyclomatic complexity when computing it: you will only have 2 cases for this condition (the ``stream`` string vs anything else) instead of having 3 cases (the ``stream`` string vs any other string vs any other type)).